.github/workflows/cd.yml

```diff
@@ -211,14 +211,13 @@ jobs:
       - name: Build and push release Docker images by digest
         id: build
         uses: docker/build-push-action@v5
+        target: release
         with:
           platforms: ${{ matrix.arch.platform }}
           build-args: |
             PACKAGE=${{ matrix.name.package }}
             TARGET=${{ matrix.arch.target }}
           context: rust
-          cache-from: |
```

Comment:
I think the Docker image caching may be a little too aggressive. This was causing the integration tests to pass or fail when they shouldn't, even after changing the Rust target triple the binaries were built for.

It might be a good idea to remove it so that others don't end up sinking a bunch of time debugging this issue, or so that we don't accidentally ship a Docker image with stale layers.

The actual binaries inside the image are replaced by the `cross build` step later on so the risk of shipping a stale `gateway` or `relay` binary is lower.

Reply:
I've since reverted the tokio-tungstenite change and will try again when `main` is green.